### PR TITLE
Lucene99HnswVectorsReader.search float-vs-byte variants: reduce code duplication

### PR DESCRIPTION
**Pull request by cpoerschke**

### CODE REVIEW

File: lucene/core/src/java/org/apache/lucene/codecs/lucene99/Lucene99HnswVectorsReader.java [hunk 1]
```diff
@@ -248,45 +248,53 @@ public ByteVectorValues getByteVectorValues(String field) throws IOException {
...
+  private abstract static class RandomVectorScorerSupplier {
+    abstract RandomVectorScorer get() throws IOException;
+  }
+
+  private void search(
+      FieldEntry fieldEntry,
+      KnnCollector knnCollector,
+      Bits acceptDocs,
+      VectorEncoding vectorEncoding,
+      RandomVectorScorerSupplier scorerSupplier)
```
Comment (benwtrent) [inline lucene/core/src/java/org/apache/lucene/codecs/lucene99/Lucene99HnswVectorsReader.java:289]: Why not use a `IOSupplier<RandomVectorScorer>` ? The added abstract class doesn't seem worth it. 

Then the calling code can be simplified to `() -> flatVectorsReader.getRandomVectorScorer(field, target)`

**Review by benwtrent — COMMENTED** [review]
I think it could be simplified a bit more with `IOSupplier` and lambdas

File: lucene/core/src/java/org/apache/lucene/codecs/lucene99/Lucene99HnswVectorsReader.java (hunk 1 above)
Comment (cpoerschke) [reply lucene/core/src/java/org/apache/lucene/codecs/lucene99/Lucene99HnswVectorsReader.java:289]: Ah, thanks! I didn't know about `IOSupplier`'s existence: https://github.com/apache/lucene/blob/releases/lucene/9.11.1/lucene/core/src/java/org/apache/lucene/util/IOSupplier.java

Will make the change to use it.

**Review by benwtrent — APPROVED** [review]
So CLEAN!

**Review by tteofili — APPROVED** [review]

**Pull request merged**